time/src/lib.rs

```diff
@@ -300,4 +300,14 @@ impl UTime {
         let datetime: DateTime<Utc> = DateTime::from_utc(naive, Utc);
         format!("{}", datetime)
     }
+
+    pub fn days_hours_mins_secs(&self) -> (u64, u64, u64, u64) {
+        let days = self.0 / (1000 * 24 * 60 * 60);
+        let hours = (self.0 - days * 1000 * 24 * 60 * 60) / (1000 * 60 * 60);
+        let mins = (self.0 - days * 1000 * 24 * 60 * 60 - hours * 1000 * 60 * 60) / (1000 * 60);
+        let secs =
+            (self.0 - days * 1000 * 24 * 60 * 60 - hours * 1000 * 60 * 60 - mins * 1000 * 60)
```

Comment:
Rule 1 : never reinvent cryptographic primitives, use libs, otherwise you WILL introduce security problems
Rule 2 : never reinvent calendar primitives, use libs, otherwise you WILL miscalculate (ex: leap seconds, precession corrections, etc...)

Reply:
It's a duration not a date so a minute is always 60 secs and so on :)

Reply:
a day is not always 24 hours ;)

Reply:
To do that we must use [datetime](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html), but from what I recall we didn't want to go down that path.

Reply:
ok then let's keep it like that...